### ISSUE TITLE
stamps are not rendered correctly

### ISSUE THREAD

**Issue by SHartmann-Roesberg** [mozilla/pdf.js#14105]
PDF file:
[Stamp_Problem.pdf](https://github.com/mozilla/pdf.js/files/7291567/Stamp_Problem.pdf)

Configuration:

- Web browser and its version: current Firefox, Chrome 
- Operating system and its version: Windows 10 
- PDF.js version: current (2.10.377), using viewer from demo site

Steps to reproduce the problem:
Open the attached pdf in a viewer.
If you open the pdf with Acrobat or with integrated Chrome PDF Viewer etc. it looks like in screenshot 1.
If you open it with pdf-js in Chrome, Firefox, Edge or use the integrated PDFViewer of Firefox it looks like screenshot 2.

expected behavior:

The stamps are rendered on top of the table and shown correctly

![image](https://user-images.githubusercontent.com/69242952/136157353-553a337d-a0f5-4e69-a574-39853a6f8b49.png)

What went wrong:

the stamps are only visible on top of the table lines

![image](https://user-images.githubusercontent.com/69242952/136157267-432a38f8-a001-4e14-bffe-462bb81d9a50.png)

**Comment by Snuffleupagus** [comment]
Duplicate of #13571, given that https://github.com/mozilla/pdf.js/issues/13571#issuecomment-877607263 applies here as well.

**Comment by Snuffleupagus** [comment]
Interestingly, if you skip the regular operatorList[1] and *only* render the Annotations this actually works correctly!
Hence it seems that we're *somehow* not correctly resetting the state after the "normal" operatorList and before the Annotation-operatorLists are handled (in the `src/display/canvas.js` file).

---
[1] By *replacing* all of https://github.com/mozilla/pdf.js/blob/2822ba42082b26fae9ecd9494c824d18ef9b44a2/src/core/document.js#L346-L355 with a `return opList;` line.

**Comment by Snuffleupagus** [comment]
> I think we can easily fix this in using the `hasOwnCanvas` flag.

That'd seem like a work-around, rather than a "real" fix, and it seems quite important (to me at least) that we figure out *why* it happens since this does suggest something being wrong w.r.t. resetting the canvas-state before rendering Annotations.

**Comment by calixteman** [comment]
The last commands on the page are:
```
0 G 0 J 0 j 1.44 w 10 M []0 d 411.84 149.76 1.68 630.96 re W n
412.8 780.48 m 412.8 149.76 l S
```
So the clip region is very small (`width = 1.68`) and then we draw the annotations.
So drawing in a new canvas in using the `hasOwnCanvas` stuff is helping to have no clipping path: it's why it works.

I think that a correct fix would be to restore (in `beginAnnotations`) the context saved when we start to draw the page in order to have something clean before starting drawing the annotations.
I suppose it doesn't really make sense to use something which has been set during the page drawing (before the annotations rendering).
@Snuffleupagus, wdyt ?
